R/Design-class.R

```diff
@@ -409,110 +409,118 @@ DualResponsesSamplesDesign <- function(eff_model,
   )
 }
 
-# nolint start
-
-## ---------------------------------------------------------------------------------------------------
-## class for design based on DLE and efficacy response without  samples using pseudo DLE and efficacy models
-## ----------------------------------------------------------------------------------------------------
-##' This is a class of design based on DLE responses using the \code{\linkS4class{LogisticIndepBeta}} model
-##' model and efficacy responses using \code{\linkS4class{ModelEff}}  model class
-##' without DLE and efficacy samples. It contain all slots in
-##' \code{\linkS4class{RuleDesign}} and \code{\linkS4class{TDDesign}} class object
-##'
-##' @slot data the data set of \code{\linkS4class{DataDual}} class object
-##' @slot Effmodel the pseudo efficacy model to be used, an object class of
-##' \code{\linkS4class{ModelEff}}
-##'
-##' @example examples/design-class-DualResponsesDesign.R
-##' @export
-##' @keywords class
+# DualResponsesDesign ----
+
+## class ----
+
+#' `DualResponsesDesign`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This is a class of design based on DLE responses using the [`LogisticIndepBeta`] model
+#' model and efficacy responses using [`ModelEff`]  model class
+#' without DLE and efficacy samples. It contain all slots in
+#' [`RuleDesign`] and [`TDDesign`] class object
+#'
+#' @slot data the data set of [`DataDual`] class object
+#' @slot eff_model the pseudo efficacy model to be used, an object class of
+#' [`ModelEff`]
+#'
+#' @export
+#'
 .DualResponsesDesign <-
   setClass(
     Class = "DualResponsesDesign",
-    representation(
-      Effmodel = "ModelEff",
+    representation = list(
+      eff_model = "ModelEff",
       data = "DataDual"
     ),
-    prototype(
+    prototype = prototype(
       nextBest = .NextBestMaxGain(),
       data = DataDual(doseGrid = 1:2),
       startingDose = 1,
       model = .LogisticIndepBeta()
     ),
-    contains = list("TDDesign")
+    contains = "TDDesign"
   )
-validObject(.DualResponsesDesign())
-
-
-##' Initialization function for 'DualResponsesDesign"
-##' @param data please refer to \code{\linkS4class{DualResponsesDesign}} class object
-##' @param Effmodel please refer to \code{\linkS4class{DualResponsesDesign}} class object
-##' @param \dots additional arguments for \code{\link{TDDesign}}
-##' @return the \code{\linkS4class{DualResponsesDesign}} class object
-##'
-##' @export
-##' @keywords methods
-DualResponsesDesign <- function(Effmodel,
+
+## constructor ----
+
+#' @rdname DualResponsesDesign-class
+#' @param data please refer to [`DualResponsesDesign`] class object
+#' @param eff_model please refer to [`DualResponsesDesign`] class object
+#' @param \dots additional arguments for \code{\link{TDDesign}}
+#' @inheritDotParams RuleDesign
+#'
+#' @example examples/design-class-DualResponsesDesign.R
```

Comment:
```suggestion
#' @example examples/Design-class-DualResponsesDesign.R
```
and please rename this file accordingly